### PR TITLE
EUI-1199

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-1199

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by udaydenduluri33** [comment]
> Can we add tests?

@andrewfolga  I have added the tests now

File: src/shared/components/palette/collection/write-collection-field.html [hunk 1]
```diff
@@ -5,7 +5,10 @@
     <h2 class="heading-h2">
       {{caseField | ccdFieldLabel}}
     </h2>
-    <button class="button" type="button" (click)="addItem(true)" [disabled]="isNotAuthorisedToCreate()">Add new</button>
+    <ng-container *ngIf="!isNotAuthorisedToCreate()">
```
Comment (MSancaktutar) [inline src/shared/components/palette/collection/write-collection-field.html]: double negation - one from "!" and another from the function name

File: src/shared/components/palette/collection/write-collection-field.html [hunk 2]
```diff
@@ -19,9 +22,10 @@ <h2 class="heading-h2 error-spacing">
             <label [for]="buildIdPrefix(i) + i"><h3 class="heading-h3">{{itemLabel(i)}}</h3></label>
           </div>
           <div class="float-right">
-            <button class="button button-secondary" type="button" (click)="openModal(i)"
-                    [disabled]="isNotAuthorisedToDelete(i)"
+              <ng-container *ngIf="!isNotAuthorisedToDelete(i)">
```
Comment (MSancaktutar) [inline src/shared/components/palette/collection/write-collection-field.html]: double negation - one from "!" and another from the function name
makes code less smooth to read

File: src/shared/components/palette/collection/write-collection-field.html [hunk 3]
```diff
@@ -40,9 +44,9 @@ <h2 class="heading-h2 error-spacing">
       </div>
 
     </div>
-
-    <button class="button" type="button" (click)="addItem()" [disabled]="isNotAuthorisedToCreate()" *ngIf="caseField.value && caseField.value.length">Add new</button>
-
+    <ng-container *ngIf="!isNotAuthorisedToCreate()">
```
Comment (MSancaktutar) [inline src/shared/components/palette/collection/write-collection-field.html]: double negation - one from "!" and another from the function name
makes code less smooth to read

**Pull request closed without merging**